loadbalancer/src/main/java/com/boundary/dropwizard/consul/loadbalancer/AbstractLBFactory.java

```diff
@@ -72,20 +76,35 @@ protected ServiceHealthCache buildCache(Environment env, HealthClient healthClie
                 getWatchSeconds()
         );
 
-        env.lifecycle().manage(new Managed() {
-            @Override
-            public void start() throws Exception {
+        if (env != null) {
+            env.lifecycle().manage(new Managed() {
+                @Override
+                public void start() throws Exception {
+                    cache.start();
+                    if (!cache.awaitInitialized(10, TimeUnit.SECONDS)) {
+                        throw new Exception("load balancer init timeout");
+                    }
+                }
+
+                @Override
+                public void stop() throws Exception {
+                    cache.stop();
+                }
+            });
+        } else {
+            try {
                 cache.start();
                 if (!cache.awaitInitialized(10, TimeUnit.SECONDS)) {
-                    throw new Exception("load balancer init timeout");
+                    LOGGER.error("load balancer init timeout without dropwizard Environment");
                 }
+            } catch (InterruptedException iE) {
+                LOGGER.error("caught InterruptedException in buildCache:");
+                iE.printStackTrace();
```

Comment:
these don't need to be printed if they are logged. it would also be better to re-throw the exception rather than just logging something and handing back a cache that won't work.